src/hooks/use-update-demo-site.tsx

```diff
@@ -76,12 +76,21 @@ export const DemoSiteUpdateProvider: React.FC< DemoSiteUpdateProviderProps > = (
 				} );
 				Sentry.captureException( error );
 			} finally {
-				setDemoSiteUpdating( false );
+				setUpdatingSites( ( prev ) => {
+					const newSet = new Set( prev );
+					newSet.delete( localSite.id );
+					return newSet;
+				} );
```

Comment:
The current implementation is good, so no need to change it. But wanted to share an alternative in case we'd like to use a string array for the state:

```suggestion
				setUpdatingSites( ( prev ) => prev.filter( ( id ) => id === localSite.id ) );
```